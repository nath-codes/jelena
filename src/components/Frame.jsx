import styled, { css } from "styled-components";
import tile from "../assets/img/tile.png";

const StyledFrame = styled.div`
  background: url(${tile}) no-repeat;
  background-size: contain;
  height: 250px;
  position: fixed;
  width: 250px;

  ${props =>
    props.position === "bottom-left" &&
    css`
      bottom: 0;
      left: 0;
      transform: rotate(270deg);
    `}

  ${props =>
    props.position === "bottom-right" &&
    css`
      bottom: 0;
      right: 0;
      transform: rotate(180deg);
    `}

  ${props =>
    props.position === "top-left" &&
    css`
      left: 0;
      top: 0;
    `}

  ${props =>
    props.position === "top-right" &&
    css`
      right: 0;
      top: 0;
      transform: rotate(90deg);
    `}

`;

export default StyledFrame;
