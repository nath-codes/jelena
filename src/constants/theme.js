export default {
  fonts: {
    heading: "bebas-neue",
    body: "montserrat"
  },
  colors: {
    delftBlue: "#1b56c5",
  }
};
