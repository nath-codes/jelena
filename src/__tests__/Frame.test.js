import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { ThemeProvider } from "styled-components";
import Frame from "../components/Frame";
import theme from "../constants/theme";

describe("<Frame />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Frame />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
