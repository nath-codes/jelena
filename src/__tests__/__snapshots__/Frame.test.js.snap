// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Frame /> renders correctly 1`] = `
.c0 {
  background-image: url(tile-xs.png);
  background-repeat: no-repeat;
  background-size: contain;
  height: 125px;
  position: fixed;
  width: 125px;
}

@media (-webkit-min-device-pixel-ratio:2) {
  .c0 {
    background-image: url(tile-xs@2x.png);
  }
}

@media (min-width:576px) {
  .c0 {
    background-image: url(tile-s.png);
    height: 150px;
    width: 150px;
  }
}

@media (min-width:576px) and (-webkit-min-device-pixel-ratio:2) {
  .c0 {
    background-image: url(tile-s@2x.png);
  }
}

@media (min-width:768px) {
  .c0 {
    background-image: url(tile-m.png);
    height: 200px;
    width: 200px;
  }
}

@media (min-width:768px) and (-webkit-min-device-pixel-ratio:2) {
  .c0 {
    background-image: url(tile-m@2x.png);
  }
}

@media (min-width:992px) {
  .c0 {
    background-image: url(tile-l.png);
    height: 250px;
    width: 250px;
  }
}

@media (min-width:992px) and (-webkit-min-device-pixel-ratio:2) {
  .c0 {
    background-image: url(tile-l@2x.png);
  }
}

<div
  className="c0"
/>
`;
